File: src/main/java/com/gpch/login/controller/PlaceController.java
package com.gpch.login.controller;


import com.gpch.login.model.Place;
import com.gpch.login.repository.PlaceRepository;
import com.gpch.login.service.PlaceService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.validation.BindingResult;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;

import javax.validation.Valid;

@Controller
@RequestMapping("/places/")
public class PlaceController {

    @Autowired
    private PlaceService placeService;

    @GetMapping("/list/")
    public String placeList(Model model){
        model.addAttribute("places", placeService.getPlacesList());
        return "place_list";
    }

    @GetMapping("/add/")
    public String showAddPlaceForm(@ModelAttribute Place place, BindingResult result, Model  model){
        return "place_add";
    }

    @PostMapping("/add/")
    public String addPlace(@Valid @ModelAttribute Place place, BindingResult result, Model  model){
        System.out.println("Dodalem miejsce");
        return "place_add";
    }
}

File: src/main/java/com/gpch/login/controller/PlaceController.kt
package com.gpch.login.controller


import com.gpch.login.dto.PlaceAddDTO
import com.gpch.login.repository.PlaceRepository
import com.gpch.login.service.PlaceService
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.validation.BindingResult
import org.springframework.web.bind.annotation.*

import javax.validation.Valid

@Controller
@RequestMapping("/places/")
class PlaceController(
        private val placeService: PlaceService,
        private val placeRepository: PlaceRepository
) {

    @GetMapping("/list/")
    fun placeList(model: Model): String {
        val placesList = placeService.placesList()
        println("PlaceList: $placesList")
        model.addAttribute("placeList",placesList )
        return "place_list"
    }

    @GetMapping("/verify_place_list/")
    fun verifyPlaceList(model: Model): String {
        val placesList = placeService.getPlacesToVerify()
        println("PlaceList: $placesList")
        model.addAttribute("placeList",placesList )

        return "admin/verify_place_list"
    }

    @GetMapping("/add/")
    fun showAddPlaceForm(@ModelAttribute placeAddDTO: PlaceAddDTO): String {
        return "place_add"
    }

    @PostMapping("/add/")
    fun addPlace(@Valid @ModelAttribute placeAddDTO: PlaceAddDTO, result: BindingResult): String {
        //wyłączona walidacja poki co
        placeService.verifyAndSavePlace(placeAddDTO, result)
        return if(result.hasErrors())
            "place_add"
        else
            return  "redirect:../list/"
    }


    @PostMapping("/verify/{id}")
    fun verify(@PathVariable id : String)  {
        val selectedPlace = placeRepository.findById(id.toInt()).get()
        selectedPlace.verified = true

    }

    @GetMapping("/{id}")
    fun showPlaceDetails(@PathVariable id: String, model : Model ): String{
        val current = placeRepository.findById(id.toInt()).get()
        model.addAttribute("model",current)
        return "place_detail"
    }

}
